test/cpp/cpp_test_util.cpp

```diff
@@ -68,6 +62,14 @@ void ForEachDevice(const std::function<void(const Device&)>& devfn) {
   devfn(Device(default_device));
 }
 
+void ForEachTorchXLADevice(const std::function<void(const torch::Device&)>& devfn) {
+  std::string default_device =
+      xla::ComputationClient::Get()->GetDefaultDevice();
+  int device_ord = std::stoi(default_device.substr(default_device.find(":")+1));
```

Comment:
This is wrong. Internal device ordinals are not matching the pytorch device ones.
You want to use something around `AtenDefaultDevice()`:

https://github.com/pytorch/xla/blob/cab477c2d17abed3f21a37693d6bb6ee9a63e2b7/torch_xla/csrc/aten_xla_bridge.cpp#L180